Tests/MongoSwiftTests/MongoDatabaseTests.swift

```diff
@@ -112,26 +112,52 @@ final class MongoDatabaseTests: MongoSwiftTestCase {
         )
 
         expect(try db.createCollection("fooView", options: viewOptions)).toNot(throwError())
-        let decoder = BSONDecoder()
-        var collectionInfo = try db.listCollections().map { try decoder.decode(CollectionInfo.self, from: $0) }
+
+        var collectionInfo = try Array(db.listCollections()) as [CollectionSpecification]
         collectionInfo.sort { $0.name < $1.name }
 
         expect(collectionInfo).to(haveCount(3))
 
-        let expectedFoo = CollectionInfo(name: "foo", type: "collection", options: fooOptions)
+        let expectedFoo = CollectionSpecification(name: "foo",
+                                                  type: CollectionType(rawValue: "collection"),
```

Comment:
you should be able to initialize a new `CollectionType` here by just saying `.collection`. based on the context the  compiler knows the type is `CollectionType` so it will look for a matching enum case there. same applies a couple lines down with `.view`.